### PR TITLE
Improved handling of small subarrays. 

### PR DESCRIPTION
**Pull request by danielczech**
- When initiating recording, ensure that all the instances that are subscribed, are also ready. `subscribed` should be a subset of `ready` (it is possible in some cases that only a subset of the available `ready` instances are subscribed, e.g. in the case of small subarrays, and so recording should proceed so long as all `subscribed` instances are also `ready`, not the other way around). 

- Pre-sort instances in case the active subarray is an unusually small one and is only using a subset of the available instances (we need to subscribe to instances in sorted physical order, otherwise packet loss could occur for other MeerKAT users). 

- Report a summary of error codes rather than simply printing out the full list.

### CODE REVIEW

File: coordinator/redis_util.py [hunk 1]
```diff
@@ -142,6 +142,13 @@ def reset_dwell(r, instances, dwell):
     for i in range(len(chan_list)):
         r.publish(chan_list[i], f"DWELL={dwell}")
 
+def channel_list(hpgdomain, instances):
+    """Build a list of Hashpipe-Redis Gateway channels from a list
+       of instance names (of format: host/instance)
+    """
+    channel_list = [hpgdomain + '://' + instance + '/set' for instance in instances]
```
Comment (lacker) [inline coordinator/redis_util.py:149]: might be a bit clearer with an f-string

File: coordinator/states.py [hunk 2]
```diff
@@ -141,7 +143,7 @@ def on_entry(self, data):
 
         subscribed = data["subscribed"]
         ready = data["ready"]
-        if ready == subscribed.intersection(ready):
+        if subscribed == subscribed.intersection(ready):
```
Comment (lacker) [inline coordinator/states.py:146]: issubset seems a bit simpler, if I understand correctly what this line is doing

**Comment by lacker** [comment]
Looks good to me, just a couple minor suggestions.

File: coordinator/states.py (hunk 2 above)
Comment (danielczech) [reply coordinator/states.py]: Agreed, I've changed it

**Pull request merged**